### PR TITLE
Add the example non-genai qnn model to ci and benchinfra

### PR DESCRIPTION
**Pull request by guangy10**

### CODE REVIEW

**Comment by guangy10** [comment]
Fix `EXECUTORCH_ROOT` env

**Comment by guangy10** [comment]
@haowhsu-quic The script to export dl3 doesn't work. Same failure when I run the export script locally from latest main. See the job https://github.com/pytorch/executorch/actions/runs/10426838811/job/28880603010?pr=4747

File: .github/workflows/trunk.yml [hunk 1]
```diff
@@ -270,3 +270,26 @@ jobs:
...
+    name: test-qnn-model
+    uses: pytorch/test-infra/.github/workflows/linux_job.yml@main
+    strategy:
+      matrix:
+        dtype: [fp32]
+        model: [dl3]
+      fail-fast: false
+    with:
+      runner: linux.2xlarge
+      docker-image: executorch-ubuntu-22.04-clang12-android
```
Comment (cccclai) [inline .github/workflows/trunk.yml:284]: @kirklandsign do we want to preinstall qnn for this docker image or have a dedicated docker image for qnn models?

File: .ci/scripts/test.sh [hunk 2]
```diff
@@ -129,9 +144,30 @@ test_model_with_xnnpack() {
...
+  source "$(dirname "${BASH_SOURCE[0]}")/build-qnn-sdk.sh"
+  echo "ANDROID_NDK_ROOT: $ANDROID_NDK_ROOT"
+  echo "QNN_SDK_ROOT: $QNN_SDK_ROOT"
+  echo "EXECUTORCH_ROOT: $EXECUTORCH_ROOT"
+
+  export LD_LIBRARY_PATH=$QNN_SDK_ROOT/lib/x86_64-linux-clang/
+  export PYTHONPATH=$EXECUTORCH_ROOT/..
+
+  if [[ "${MODEL_NAME}" == "dl3" ]]; then
+    "${PYTHON_EXECUTABLE}" -m examples.qualcomm.scripts.deeplab_v3 -b ${CMAKE_OUTPUT_DIR} -m SM8550 --compile_only --download
```
Comment (cccclai) [inline .ci/scripts/test.sh:157]: Is this test expected to just generate the model or including running the model?

**Comment by haowhsu-quic** [comment]
> @haowhsu-quic The script to export dl3 doesn't work. Same failure when I run the export script locally from latest main. See the job https://github.com/pytorch/executorch/actions/runs/10426838811/job/28880603010?pr=4747

Hi, I think this was a breakage introduced by recent PR https://github.com/pytorch/executorch/pull/4566.
Would it be possible to have https://github.com/pytorch/executorch/pull/4774 for resolving issue?

File: .github/workflows/trunk.yml (hunk 1 above)
Comment (kirklandsign) [reply .github/workflows/trunk.yml:284]: We don’t have that right now.

File: .ci/scripts/test.sh (hunk 2 above)
Comment (kirklandsign) [reply .ci/scripts/test.sh:157]: Seems it’s AOT right now.

File: .github/workflows/trunk.yml (hunk 1 above)
Comment (guangy10) [reply .github/workflows/trunk.yml:284]: @cccclai Moving the deps to docker can be done later. It shouldn't be a blocker for enabling qnn delegation in the benchmarking workflow

**Comment by guangy10** [comment]
Rebased after #4774 got merged

**Comment by guangy10** [comment]
`test-qnn-model` job passes

**Comment by guangy10** [comment]
Now we can see the qnn and xnnpack models are exported, uploaded and deployed to S22 via device farm successfully: https://github.com/pytorch/executorch/actions/runs/10463149566. Once we have 1) the generic apk loader and 2) bundled models we should be able to benchmark these models ondevice

Only dl3 is added to the CI and benchinfra, other models are missing instructions to use. See #4767.

**Review by metascroy — APPROVED** [review]

**Review by kirklandsign — APPROVED** [review]

**Pull request merged**